custom/intrahealth/utils.py

```diff
@@ -360,8 +360,14 @@ def get_report_context(self, data_provider):
 
     @property
     def export_table(self):
-        reports = [r['report_table'] for r in self.report_context['reports']]
-        return [self._export_table(r['title'], r['headers'], r['rows'], total_row=r['total_row']) for r in reports]
+        export_tables = []
+        for individual_report in self.report_context['reports']:
+            report_table = individual_report['report_table']
+            # The keys contain html info, so only the values get exported to excel
+            total_row = [row_value.items()[0][1] for row_value in report_table['total_row']]
```

Comment:
One possible benefit of going the function approach is you could also test and handle any possibly incorrect assumptions you're making about what the expected format of those things are. E.g. if you do just get a string you can easily return that instead of extracting the actual value from the more complex format. I think this could also then avoid the need to make sure you manually check each report.

Reply:
Good call, I separated it into a function and let it accept inputs for non-dicts and added some tests. Not sure if this is what you mean avoid the need for manually check each report, but it would stop checking each entry after seeing that the first entry isn't an html dict. Do you know if the other format I'm now checking for (a list of strings) is the standard format that these tables will come in as?